src/controller/include/controller/recording_controller.h

```diff
@@ -0,0 +1,45 @@
+#include "controller/controller.h"
+
+#include <array>
+#include <bitset>
+#include <fstream>
+#include <jsoncpp/json/json.h>
+#include <jsoncpp/json/writer.h>
+
+#pragma once
+
+namespace controller {
+class RecordingController : public Controller {
+	/**
+	 * Output Stream to write to Json File
+	 */
+	std::ofstream json_file;
+
+	/**
+	 * Stores the Json value to write to Json file
+	 */
+	Json::Value frame_button_value;
+
+	/**
+	 * Keep a track a ticks
+	 */
+	unsigned long long ticks = 0;
+
+  public:
+	/**
+	 * @see ControllerInterface#press_button
+	 * Record JSON inputs here
+	 */
+	void press_button(Button button) override;
```

Comment:
Why only press button and not release button?

You'll need to record the button releases as well, might as well do that now. You can make that part of the JSON format.

Reply:
Yeah okay, this makes sense actually.

Reply:
But again, we'll have to keep a `release_flag` in `json_file` which will indicate which key to press and which to release.

Reply:
That sounds alright to me. Encode whatever information is needed in the JSON unless there's a reason not to.

Reply:
Cool then!